fix(hpExtension):Match the updated cs3d setPan/setZoom calls

wayfarer3130 opened the pull request:
The CS3D options changed so that the entire display area/canvas center functionality is now embedded in with the setPan/setZoom calls rather than having separate options.

wayfarer3130 committed:
fix(hpExtension):Match the updated cs3d setPan/setZoom calls

wayfarer3130 committed:
Fix the setViewportZoomPan to work to PR updates

wayfarer3130 commented on extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts:
```diff
@@ -5,30 +5,67 @@
-  if( zoom && zoom!==1 ) viewport.setZoom(zoom);
-  if( pan ) viewport.setPan(pan);
-  if( flipHorizontal || flipVertical ) {
+const setViewportZoomPan = (id, value, viewport) => {
```
Need to apply the imageCanvasPoint and initialDisplay area manually due to PR changes on CS3D

wayfarer3130 commented on extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts:
```diff
@@ -5,30 +5,67 @@
+
+  const imageData = viewport.getImageData();
+
+  if (imageCanvasPoint && imageData) {
```
This code works like the old version, but still has a few bugs - not sure why yet - seems like the canvas size isn't quite right, or the center point isn't actually the center of the image area.

wayfarer3130 commented on extensions/hp-extension/src/hp/hpBreast.ts:
```diff
@@ -134,10 +137,9 @@ export default {
             background: [127,127,0],
-            initialCenter: [0,0],
-            canvasCenter: [0,0],
             customViewportOptions: {
```
Needed to move everything into the zoomPan object

ladeirarodolfo commented on extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts:
```diff
@@ -5,30 +5,67 @@
-  if( flipHorizontal || flipVertical ) {
+const setViewportZoomPan = (id, value, viewport) => {
+  const { zoom, pan, flipHorizontal, flipVertical } = value;
+  const { imageCanvasPoint, initialDisplayArea } = value;
```
join imageCanvasPoint, initialDisplayArea  to { zoom, pan, flipHorizontal, flipVertical }

ladeirarodolfo commented on extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts:
```diff
@@ -5,30 +5,67 @@
+   
+    const canvasZero = viewport.worldToCanvas([0,0,0]);
+    const canvasEdge = viewport.worldToCanvas(imageData.dimensions);
+    const canvasImage = [canvasEdge[0]-canvasZero[0], canvasEdge[1]-canvasZero[1]];
```
there is no need on creating an array to later do a destructuring.

WE can just define imageWidth directly, no? const imageWidth = canvasEdge[0]-canvasZero[0];

ladeirarodolfo commented on extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts:
```diff
@@ -5,30 +5,67 @@
+  // Now set the sync value offsets
+  if (zoom && zoom !== 1) viewport.setZoom(zoom);
+  if (pan) {
+    console.log("Setting pan to", JSON.stringify(pan));
```
Do we really need to log this? Why not on Zoom?

ladeirarodolfo commented on extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts:
```diff
@@ -5,30 +5,67 @@
+    const canvasEdge = viewport.worldToCanvas(imageData.dimensions);
+    const canvasImage = [canvasEdge[0]-canvasZero[0], canvasEdge[1]-canvasZero[1]];
+    const [imgWidth,imgHeight] = canvasImage;
+    const imagePanX = imgWidth * (0.5-imageCanvasPoint[0]);
```
Ive seen some prettierfy incosistence here and some over this file. Is lint/prettier being applied at all?

wayfarer3130 replied:
No, still need to setup lint appropriately on the files.  Will do it on master, and then apply the lint everywhere else.

ladeirarodolfo commented on extensions/hp-extension/src/hp/hpBreast.ts:
```diff
@@ -152,13 +154,12 @@ export default {
                 flipVertical: true,
               },
-            },
+            },            
```
remove extra space please

ladeirarodolfo reviewed (changes requested):
Thanks for your PR Bill,

I have added only minor cosmetic changes.
Please also ensure your linter/prettier runs over all changed files

wayfarer3130 committed:
Fix HP for latest version

wayfarer3130 committed:
fix(hp-synchronizers):Add support for the synchronizer position

This update changes the zoom/pan synchronization to work with the
OHIF v3-stable (soon to be merged) changes for the method to register
synchronization and initial display settings.  This includes both
temporal synchronization (previous/next), and initial sync to the
initial values.

ladeirarodolfo commented on extensions/hp-extension/src/synchronizers/initialZoomPanCallback.ts:
```diff
@@ -0,0 +1,65 @@
+}
+
+export function storeCurrentZoomPan(synchronizerInstance: Synchronizer,  sourceViewport: Types.IViewportId) {
+  const renderingEngine = getRenderingEngine(sourceViewport.renderingEngineId);
```
This method is being tied to when ELEMENT is disabled. I'm wondering if getRenderingEngine and some other engine methods will work due to Element being ALREADY disabled

ladeirarodolfo commented on extensions/hp-extension/src/synchronizers/createSingleViewportSynchronizer.ts:
```diff
@@ -0,0 +1,43 @@
+  console.log("Add viewport info called for", viewportId);
+}
+
+function addEvent(eventName, listener) {
```
I would recommend to implement also the removeEvent

ladeirarodolfo commented on extensions/hp-extension/src/synchronizers/createInitialZoomPanSynchronizer.ts:
```diff
@@ -0,0 +1,24 @@
+ * @param synchronizerName - The name of the synchronizer.
+ * @returns A new `Synchronizer` instance.
+ */
+export default function createSynchronizer(
```
do we have the ability to create, but what about destroy?

ladeirarodolfo commented on extensions/hp-extension/src/synchronizers/createInitialZoomPanSynchronizer.ts:
```diff
@@ -0,0 +1,24 @@
+ * @returns A new `Synchronizer` instance.
+ */
+export default function createSynchronizer(
+  synchronizerName: string,
```
What happens if by mistake we create a second synchroniser with same name?

ladeirarodolfo commented on extensions/hp-extension/src/index.tsx:
```diff
@@ -23,14 +24,16 @@ export default {
     HangingProtocolService.addCustomAttribute('seriesDescriptions', 'Series Descriptions', seriesDescriptionsFromDisplaySets);
     HangingProtocolService.addCustomAttribute('numberOfDisplaySets', 'Number of displays sets', numberOfDisplaySets);
+  
+    SyncGroupService.setSynchronizer('initialzoompan', initialZoomPan);
```
It seems there is a very specific version, that exposes servicesManager.services.SyncGroupService. setSynchronizer.
We again come to the coupling thing here. I was wondering if we could at least have a conditional to ensure there is this  method.

ladeirarodolfo commented on extensions/hp-extension/src/hp/hpHeart.ts:
```diff
@@ -245,11 +230,14 @@ export default {
+              {
+                type: 'initialZoomPan',
+                options: { 
+                  initialDisplayArea: [0.4, 0.4],
```
there is a inconsistence on your linter. Some places like here there is a space. Some other there is not. Can you ensure your linter or project linter runs on all files?

wayfarer3130 committed:
feat(plugin):Loads the extensions as plugins instead of link

wayfarer3130 committed:
Allow public publishing

wayfarer3130 committed:
Fixing publish list

wayfarer3130 committed:
v3.0.3

wayfarer3130 committed:
fix(ecg):Display ecg on older OHIF without the retrieve bulkdata

wayfarer3130 committed:
fix(ecg):Retrieve the url from the global config and the data

wayfarer3130 committed:
Update version number

wayfarer3130 committed:
feat(microscopy):Microscopy viewer compatible with stand alone OHIF
